homeassistant/components/androidtv/media_player.py

```diff
@@ -168,180 +150,104 @@
 }
 
 
-def setup_androidtv(hass, config):
-    """Generate an ADB key (if needed) and load it."""
-    adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))
-    if CONF_ADB_SERVER_IP not in config:
-        # Use "adb_shell" (Python ADB implementation)
-        if not os.path.isfile(adbkey):
-            # Generate ADB key files
-            keygen(adbkey)
-
-        # Load the ADB key
-        signer = ADBPythonSync.load_adbkey(adbkey)
-        adb_log = f"using Python ADB implementation with adbkey='{adbkey}'"
-
-    else:
-        # Use "pure-python-adb" (communicate with ADB server)
-        signer = None
-        adb_log = f"using ADB server at {config[CONF_ADB_SERVER_IP]}:{config[CONF_ADB_SERVER_PORT]}"
-
-    return adbkey, signer, adb_log
-
-
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Android TV / Fire TV platform."""
-    hass.data.setdefault(ANDROIDTV_DOMAIN, {})
-
-    address = f"{config[CONF_HOST]}:{config[CONF_PORT]}"
 
-    if address in hass.data[ANDROIDTV_DOMAIN]:
-        _LOGGER.warning("Platform already setup on %s, skipping", address)
-        return
-
-    adbkey, signer, adb_log = await hass.async_add_executor_job(
-        setup_androidtv, hass, config
-    )
+    host = config[CONF_HOST]
 
-    aftv = await setup(
-        config[CONF_HOST],
-        config[CONF_PORT],
-        adbkey,
-        config.get(CONF_ADB_SERVER_IP, ""),
-        config[CONF_ADB_SERVER_PORT],
-        config[CONF_STATE_DETECTION_RULES],
-        config[CONF_DEVICE_CLASS],
-        10.0,
-        signer,
-    )
+    # get main data
+    config_data = {
+        CONF_HOST: host,
+        CONF_NAME: config[CONF_NAME],
+        CONF_DEVICE_CLASS: config[CONF_DEVICE_CLASS],
+        CONF_PORT: config[CONF_PORT],
+        CONF_ADB_SERVER_PORT: config[CONF_ADB_SERVER_PORT],
```

Comment:
```suggestion
        CONF_DEVICE_CLASS: config.get(CONF_DEVICE_CLASS),
        CONF_PORT: config.get(CONF_PORT),
        CONF_ADB_SERVER_PORT: config.get(CONF_ADB_SERVER_PORT),
```

Reply:
Done, also fixed the part related to the options.